File: Benchmarking/src/gen_111_27.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s1 {
    var p0: s0
    var p1: [s0]
  }
  @inline(never)
  func f0(_ v0: s1, _ v1: [[Double]], _ v2: Double) -> Double {
    let v7: [Double] = v1[0]
    let v9: [Double] = v1[0]
    let v3: Double = v9[0]
    let v10: Double = v9[0]
    let v11: Double = v7[0]
    let v6: Double = v7[0]
    let v29: [Double] = v1[0]
    let v69: Double = v29[0]
    var v37: [Double] = v7
    v37[0] = v3
    var v43: Double = v6
    let v57: Double = v37[0]
    let v116: Double = v57 / v11
    v37[0] = v43
    v37[0] = v69
    var v79: Double = v10
    v37[0] = v79
    return v116
  }
  func benchmark() {
    let v0: s1 = s1(p0: s0(p0: [[0.0], [1.0]], p1: [[2.0], [3.0], [4.0]]), p1: [s0(p0: [[5.0], [6.0]], p1: [[7.0], [8.0], [9.0]])])
    let v1: [[Double]] = [[10.0]]
    let v2: Double = 11.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 11.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()

File: Benchmarking/src/gen_111_27.scala
import java.lang.System.nanoTime
import scala.collection.immutable.Vector
object Gen extends App {
  case class s0 (
    p0: Vector[Vector[Double]],
    p1: Vector[Vector[Double]]
  )
  case class s1 (
    p0: s0,
    p1: Vector[s0]
  )
  @noinline
  def f0(v0: s1, v1: Vector[Vector[Double]], v2: Double): Double = {
    val v7: Vector[Double] = v1(0)
    val v9: Vector[Double] = v1(0)
    val v3: Double = v9(0)
    val v10: Double = v9(0)
    val v11: Double = v7(0)
    val v6: Double = v7(0)
    val v29: Vector[Double] = v1(0)
    val v69: Double = v29(0)
    var v37: Vector[Double] = v7
    v37 = v37.updated(0, v3)
    var v43: Double = v6
    val v57: Double = v37(0)
    val v116: Double = v57 / v11
    v37 = v37.updated(0, v43)
    v37 = v37.updated(0, v69)
    var v79: Double = v10
    v37 = v37.updated(0, v79)
    v116
  }
  def benchmark(): Unit = {
    val v0: s1 = s1(s0(Vector(Vector(0.0), Vector(1.0)), Vector(Vector(2.0), Vector(3.0), Vector(4.0))), Vector(s0(Vector(Vector(5.0), Vector(6.0)), Vector(Vector(7.0), Vector(8.0), Vector(9.0)))))
    val v1: Vector[Vector[Double]] = Vector(Vector(10.0))
    val v2: Double = 11.0
    val start = nanoTime()
    var result: Double = 11.0
    (1 to 1000).foreach { _ =>
      result = f0(v0, v1, v2)
    }
    val end = nanoTime()
    println(result)
    println(end - start)
  }
  benchmark()
}